src/main/java/ch/njol/skript/effects/EffCancelDrops.java

```diff
@@ -0,0 +1,112 @@
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+package ch.njol.skript.effects;
+
+import org.bukkit.event.Event;
+import org.bukkit.event.block.BlockBreakEvent;
+import org.bukkit.event.entity.EntityDeathEvent;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.ScriptLoader;
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Events;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+
+@Name("Cancel Drops")
+@Description("Cancels drops of items or experiences in a death or block break event. " +
+		"Please note that this doesn't keep items or experiences of a dead player. If you want to do that, " +
+		"use the <a href='effects.html#EffKeepInventory'>Keep Inventory / Experience</a> effect.")
+@Examples({"on death of a zombie:",
+		"\tif name of the entity is \"&cSpecial\":",
+		"\t\tcancel drops of items",
+		"",
+		"on break of a coal ore:",
+		"\tcancel the experience drops"})
+@Since("INSERT VERSION")
+@RequiredPlugins("1.12.2 or newer (cancelling item drops of blocks)")
+@Events({"death", "break / mine"})
+public class EffCancelDrops extends Effect {
+
+	private static final boolean CAN_CANCEL_BLOCK_ITEM_DROPS = Skript.methodExists(BlockBreakEvent.class, "setDropItems", boolean.class);
+
+	static {
+		Skript.registerEffect(EffCancelDrops.class,
+			"(cancel|clear|delete) [the] drops [of (1¦items|2¦[e]xp[erience][s])]",
+			"(cancel|clear|delete) [the] (1¦item|2¦[e]xp[erience]) drops");
+	}
+
+	private boolean cancelItems, cancelExps;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		cancelItems = parseResult.mark == 0 || parseResult.mark == 1;
+		cancelExps = parseResult.mark == 0 || parseResult.mark == 2;
+		if (ScriptLoader.isCurrentEvent(BlockBreakEvent.class)) {
+			if (cancelItems && !CAN_CANCEL_BLOCK_ITEM_DROPS) {
+				Skript.error("Cancelling drops of items in a block break event requires Minecraft 1.12 or newer");
+				return false;
+			}
+		} else if (!ScriptLoader.isCurrentEvent(EntityDeathEvent.class)) {
+			Skript.error("The cancel drops effect can't be used outside of a death" +
+				(CAN_CANCEL_BLOCK_ITEM_DROPS ? " or block break" : "") + " event");
+			return false;
+		}
+		if (isDelayed.isTrue()) {
+			Skript.error("Can't cancel the drops anymore after the event has already passed");
```

Comment:
I think `Can't clear the drops after the event has already passed` would be better, people would normally use `clear` instead of cancel after all.

Reply:
But "cancel" makes more sense than using "clear" *because it cancels* and i try to use the most sensible message, but it is not even an important, wrong or user code dependent thing at the end..